generate_defaults.py

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env python
```

Comment:
move this into a tools or scripts directory since it's not a core piece of the plugin.